openshift/dynamic/client.py

```diff
@@ -0,0 +1,387 @@
+#!/usr/bin/env python
+
+import sys
+import json
+from functools import partial
+
+import yaml
+from pprint import pformat
+
+from kubernetes import config
+from kubernetes.client.api_client import ApiClient
+from kubernetes.client.rest import ApiException
+
+
+class Resource(object):
+
+    def __init__(self, prefix=None, group=None, apiversion=None, kind=None,
+                 namespaced=False, verbs=None, name=None, preferred=False, client=None,
+                 singularName=None, shortNames=None, categories=None, **kwargs):
+
+        if None in (apiversion, kind):
+            raise Exception("At least kind and apiversion must be provided")
+
+        self._prefix = prefix
+        self.group = group
+        self.apiversion = apiversion
+        self.kind = kind
+        self.namespaced = namespaced
+        self.verbs = verbs
+        self.name = name
+        self.preferred = preferred
+        self.client = client
+        self.singular_name = singularName
+        self.short_names = shortNames
+        self.categories = categories
+
+        self.extra_args = kwargs
+
+    def __repr__(self):
+        if self.group:
+            groupversion = '{}/{}'.format(self.group, self.apiversion)
+        else:
+            groupversion = self.apiversion
+        return '<{}({}.{}>)'.format(self.__class__.__name__, groupversion, self.kind)
+
+    @staticmethod
+    def make_resource(prefix, group, apiversion, resource, preferred=False, client=None):
```

Comment:
Could you explain the difference between a K8sResource and OpenshiftResource? This is likely more for my own education.